frontendUiTable/src/main/java/com/virtuslab/gitmachete/frontend/ui/table/GitMacheteGraphTableManager.java

```diff
@@ -99,26 +110,25 @@ private void refreshGraphTable(Path macheteFilePath, boolean isMacheteFilePresen
     GuiUtils.invokeLaterIfNeeded(gitMacheteGraphTable::updateUI, ModalityState.NON_MODAL);
   }
 
-  private void subscribeToVcsRootChanges() {
-    // The method reference is invoked when user changes repository in dropdown menu
-    selectionChangeObservable.addObserver(this::updateAndRefreshInBackground);
-  }
-
-  private Path getMacheteFilePath(Path repoRootPath) {
-    return Try.of(() -> GitUtils.getGitDirectoryPathByRepoRootPath(repoRootPath).resolve("machete"))
-        .onFailure(e -> LOG.error("Unable to get machete file path", e))
-        .get();
+  private Path getMacheteFilePath(GitRepository gitRepository) {
+    // Using the deprecated `GitRepository#getGitDir` and not `GitRepository#getRepositoryFiles`
+    // since the latter apparently doesn't allow to extract git dir
+    // (just the paths of specific predefined files within the git dir).
+    Path gitDirPath = Paths.get(gitRepository.getGitDir().getPath());
```

Comment:
Maybe not the best solution but IMO better than using deprecated API:
```suggestion
    VirtualFile vfGitDir = GitUtil.findGitDir(gitRepository.getRoot());
    assert vfGitDir != null : "Can't get .git dir from repo root path";
    Path gitDirPath = Paths.get(vfGitDir.getPath());
```
This `GitUtil` is from `com.intellij.ui` package, so we are not dependent on any of our other modules

Reply:
Done